src/Symfony/Component/Security/Guard/GuardAuthenticatorInterface.php

```diff
@@ -83,6 +83,8 @@ public function getUser($credentials, UserProviderInterface $userProvider);
      *
      * @param mixed         $credentials
      * @param UserInterface $user
+     * 
+     * @return bool
```

Comment:
Any method might return anything. Welcome to php. :wink:

The reason behind letting the authentication fail if something other than `true` is returned, is that something might have gone wrong inside the authenticator code and you'd rather lock someone out because of a bug than letting people in who shouldn't get in. There has been a discussion on this, I just can't find it right now.

But the return value _should_ be a boolean. I could live with the change @dunglas suggested, but I'd prefer not to change it. Thoughts?